vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,653 @@
+    DeadlineBuilder::new(&[0, 1]).verify(&deadlines);
+
+    deadlines = assign_sectors_setup(Deadlines::new(), &seq(0, 15));
+    DeadlineBuilder::new(&[0, 4, 4, 4, 3]).verify(&deadlines);
+
+    deadlines = assign_sectors_setup(
+        Deadlines::new(),
+        &seq(0, (WPOST_PERIOD_DEADLINES - 1) * NEW_SECTOR_PART_SIZE + 1),
+    );
+    DeadlineBuilder::new(&[])
+        .add_to_all_from(1, NEW_SECTOR_PART_SIZE)
+        .add_to(1, 1)
+        .verify(&deadlines);
+}
+
+#[test]
+fn incremental_assignment_test() {
+    // Add one sector at a time.
+    let mut deadlines = DeadlineBuilder::new(&[0, 1]).deadlines;
+    deadlines = assign_sectors_setup(deadlines, &seq(1, 1));
+    DeadlineBuilder::new(&[0, 2]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(2, 1));
+    DeadlineBuilder::new(&[0, 3]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(3, 1));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(4, 1));
+    DeadlineBuilder::new(&[0, 4, 1]).verify(&deadlines);
+    // Add one partition at a time.
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines, &seq(0, 4));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(4, 4));
+    DeadlineBuilder::new(&[0, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(2 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines, &seq(3 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4, 4]).verify(&deadlines);
+    // Add lots
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines, &seq(0, 2 * NEW_SECTOR_PART_SIZE + 1));
+    DeadlineBuilder::new(&[
+        0,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        1,
+    ])
+    .verify(&deadlines);
+    deadlines = assign_sectors_setup(
+        deadlines.clone(),
+        &seq(2 * NEW_SECTOR_PART_SIZE + 1, NEW_SECTOR_PART_SIZE),
+    );
+    DeadlineBuilder::new(&[
+        0,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        1,
+    ])
+    .verify(&deadlines);
+}
+
+#[test]
+fn fill_partial_partitions_first_test() {
+    let mut b = DeadlineBuilder::new(&[0, 4, 3, 1]);
+    let mut deadlines = assign_sectors_setup(b.deadlines, &seq(b.next_sector_idx, 4));
+    DeadlineBuilder::new(&[0, 4, 3, 1])
+        .add_to(2, 1)
+        .add_to(3, 3)
+        .verify(&deadlines);
+
+    b = DeadlineBuilder::new(&[0, 9, 8, 7, 4, 1]);
+    deadlines = assign_sectors_setup(b.deadlines, &seq(b.next_sector_idx, 7));
+    DeadlineBuilder::new(&[0, 9, 8, 7, 4, 1])
+        .add_to(1, 3)
+        .add_to(3, 1)
+        .add_to(5, 3)
+        .verify(&deadlines);
+
+    // fill less full deadlines first
```

Comment:
in spec-actors, they have this as a seperate test case

Reply:
Does that make a difference or is it just preference?

Reply:
I mean its not a BIG deal, but they two different test cases.